css/css-text/overflow-wrap/overflow-wrap-break-word-003.html

```diff
@@ -1,18 +1,18 @@
 <!DOCTYPE html>
 <meta charset="utf-8">
-<title>CSS Text Test: overflow-wrap: break-word+break-spaces</title>
+<title>CSS Text Test: overflow-wrap:break-word + white-space:break-spaces</title>
 <link rel="author" title="Florian Rivoal" href="https://florian.rivoal.net/">
 <link rel="help" href="https://drafts.csswg.org/css-text-3/#valdef-overflow-wrap-break-word">
-<link rel="help" href="https://drafts.csswg.org/css-text-3/#valdef-overflow-wrap-break-spaces">
+<link rel="help" href="https://drafts.csswg.org/css-text-3/#valdef-white-space-break-spaces">
 <meta name="flags" content="ahem">
 <link rel="match" href="reference/overflow-wrap-break-word-003-ref.html">
 <meta name="assert" content="break-word + break-spaces do allow a break
```

Comment:
I'm a bit confused by this test case. The spec prose about the break-spaces value states that "a breaking opportunity exists AFTER every preserved white-space character. Where in the spec is stated that it can break BEFORE ? I guess that behavior is part of the overflow-wrap: break-word, but what's the effect of the white-space value in this case ? Is this test verifying a NEW braking opportunity that didin't exist before adding the 'break-spaces' value ?

Reply:
> Is this test verifying a NEW breaking opportunity that didin't exist before adding the 'break-spaces' value ?

Yes.
Also, if you have white-space:pre-wrap + overflow-wrap: break-word, you would not have a breaking opportunity, since the spaces would be collapsed (or hanged). You need white-space:break-spaces to make sure that the spaces are not collapsed/hanged, and you need overflow-wrap:break-word to add the wrapping opportunity before the first space (otherwise, it is just after).